lib/ocpp/v201/connectivity_manager.cpp

```diff
@@ -265,5 +333,97 @@ void ConnectivityManager::next_network_configuration_priority() {
         (this->network_configuration_priority + 1) % (network_connection_priorities.size());
 }
 
+void ConnectivityManager::on_websocket_connected([[maybe_unused]] int security_profile) {
+    const int actual_configuration_slot = get_active_network_configuration_slot();
+    std::optional<NetworkConnectionProfile> network_connection_profile =
+        this->get_network_connection_profile(actual_configuration_slot);
+
+    if (this->websocket_connected_callback.has_value() and network_connection_profile.has_value()) {
+        this->websocket_connected_callback.value()(actual_configuration_slot, network_connection_profile.value());
+    }
+}
+
+void ConnectivityManager::on_websocket_disconnected() {
+    int actual_configuration_slot =
+        std::stoi(ocpp::get_vector_from_csv(this->device_model.get_value<std::string>(
+                                                ControllerComponentVariables::NetworkConfigurationPriority))
+                      .at(this->network_configuration_priority));
+    std::optional<NetworkConnectionProfile> network_connection_profile =
+        this->get_network_connection_profile(actual_configuration_slot);
+
+    if (this->websocket_disconnected_callback.has_value() and network_connection_profile.has_value()) {
+        this->websocket_disconnected_callback.value()(actual_configuration_slot, network_connection_profile.value());
+    }
+}
+
+void ConnectivityManager::reconnect(WebsocketCloseReason reason, std::optional<int> next_priority) {
+    this->websocket_timer.timeout(
+        [this, reason, next_priority]() {
+            if (reason != WebsocketCloseReason::ServiceRestart) {
+                if (!next_priority.has_value()) {
+                    this->next_network_configuration_priority();
+                } else {
+                    this->network_configuration_priority = next_priority.value();
+                }
+            }
+            this->start();
+        },
+        WEBSOCKET_INIT_DELAY);
```

Comment:
Why does the reconnect always have a timeout of 2 seconds?

Reply:
No sure, this was already present (see deleted line 203in this file), so I just left it as-is.

Reply:
Yes but now you do it always when reconnecting and I don't know if that is necessary. 

@AssemblyJohn or @hikinggrass do you know maybe?

Reply:
The websocket reconnect interval is calculated in the 'get_reconnect_internal' here: https://github.com/EVerest/libocpp/blob/0c5aac4d8f2a3e3512ea3233e89810b39247b082/lib/ocpp/common/websocket/websocket_base.cpp#L149

Reply:
Yes but was there another reason to add 2 seconds to that, for example because of the old websocket implementation?
the reconnect_backoff_ms is still calculated and used when no connection could be made.

Reply:
I am sorry, for the initial delay I don't know what is with those 2 seconds init delay. It is not related to the internal websocket functionality that I had to handle. Run a git blame to see who is responsible for that delay code.

Reply:
Git blame results in it being added as part of https://github.com/EVerest/libocpp/pull/129, but I cannot fine any explanation as of why.